test/Constraints/result_builder_diags.swift

```diff
@@ -286,6 +286,21 @@ struct MyTuplifiedStruct {
   }
 }
 
+func test_invalid_return_type_in_body() {
+  tuplify(true) { _ -> (Void, Int) in
+    tuplify(false) { condition in
+      if condition {
+        return 42 // expected-error {{application of result builder 'TupleBuilder' disabled by explicit 'return' statement}}
```

Comment:
Should the error message say something different if there isn't an option to "disable" the result builder? Something like `result builder [closure|function] cannot have explicit return statement`?

Reply:
Sure, I think we can use `cannot use explicit return statement in the body of result builder %0`, WDYT?

Reply:
I like that a lot more than my suggestion, thanks!